homework_07_03_2025/BorDataStructure/BorDataStructure/Bor.cs

```diff
@@ -0,0 +1,171 @@
+        {
+            this.Size++;
+            return true;
+        }
+
+        return false;
+    }
+
+    /// <summary>
+    /// Presence of a word in a structure.
+    /// </summary>
+    /// <param name="element">String to check.</param>
+    /// <returns>Returns true if such a string exists.</returns>
+    public bool Contains(string element)
+    {
+        if (string.IsNullOrEmpty(element))
+        {
+            return false;
+        }
+
+        var len = element.Length;
+        var node = this.root;
+        for (var i = 0; i < len; i++)
+        {
+            if (!node.Nodes.ContainsKey(element[i]) || (i == len - 1 && !node.Nodes[element[i]].EndOfWord))
+            {
+                return false;
+            }
+
+            node = node.Nodes[element[i]];
+        }
+
+        return true;
+    }
+
+    /// <summary>
+    /// Remove word from structure.
+    /// </summary>
+    /// <param name="element">String to remove.</param>
+    /// <returns>Returns true if such a row existed.</returns>
+    public bool Remove(string element)
+    {
+        if (string.IsNullOrEmpty(element))
+        {
+            return false;
+        }
+
+        return RecursiveRemove(element, this.root, 0);
+    }
+
+    /// <summary>
+    /// Number of lines that start with the given prefix.
+    /// </summary>
+    /// <param name="prefix">Prefix to check.</param>
+    /// <returns>Returns number of line thath start.</returns>
+    public int HowManyStartsWithPrefix(string prefix)
+    {
+        if (string.IsNullOrEmpty(prefix))
+        {
+            return 0;
+        }
+
+        var len = prefix.Length;
+        var node = this.root;
+        for (var i = 0; i < len; i++)
+        {
+            if (!node.Nodes.ContainsKey(prefix[i]))
+            {
+                return 0;
+            }
+
+            node = node.Nodes[prefix[i]];
+        }
+
+        return node.NumberOfWordsAfterPrefix;
+    }
+
+    private static bool RecursiveAddition(string element, Node node, int index, ref bool missingLine)
+    {
+        if (index < element.Length && !node.Nodes.ContainsKey(element[index]))
+        {
+            missingLine = true;
+            node.Nodes[element[index]] = new Node();
+        }
+        else if (index == element.Length && !node.EndOfWord)
+        {
+            node.EndOfWord = true;
+            missingLine = true;
+        }
+
+        if (index < element.Length && RecursiveAddition(element, node.Nodes[element[index]], index + 1, ref missingLine))
+        {
+            node.NumberOfWordsAfterPrefix++;
+        }
+
+        return missingLine;
+    }
+
+    private static bool RecursiveRemove(string element, Node node, int index)
+    {
+        if (index == element.Length && !node.EndOfWord)
+        {
+            return false;
+        }
+        else if (index == element.Length && node.EndOfWord)
+        {
+            node.EndOfWord = false;
+            return true;
+        }
```

Comment:
Эти два `if`-а выглядят немного странно.
```suggestion
        if (index == element.Length)
        {
            if (node.EndOfWord) {
                node.EndOfWord = false;
                return true;
            } else {
                return false;
            }
        }
```